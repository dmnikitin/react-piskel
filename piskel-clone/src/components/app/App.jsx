import React from 'react';
import { Provider } from 'react-redux';
import './App.scss';
import store from '../../state/store';
import Mainbox from '../mainbox/mainbox.jsx';
import { saveToLocalStorage } from '../../helpers/localStorageHandler';

function App() {
  return (
    <Provider store={store}>
      <div className="App">
        <header className="App-header">
          <h3>PISKEL</h3>
        </header>
        <Mainbox />
      </div>
    </Provider>
  );
}

export default App;

// store.subscribe(() => saveToLocalStorage(store.getState()));
