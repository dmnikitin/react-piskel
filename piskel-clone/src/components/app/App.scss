.App {
  text-align: center;
  display: grid;
  grid-template-rows: 10% 90%;
  height: 100vh;
}

.App-header {
  background-color: #282c34;
  // min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}