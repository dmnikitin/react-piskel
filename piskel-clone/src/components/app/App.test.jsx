import React from 'react'
import { shallow, mount } from 'enzyme';
import { App } from './App';

const wrapperShallow = shallow(<App />);
const wrapperMount = mount(<App />);

describe('App component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it("displays initial div", () => {
    expect(wrapperMount.find("div")).toHaveLength(4);
  });
});

