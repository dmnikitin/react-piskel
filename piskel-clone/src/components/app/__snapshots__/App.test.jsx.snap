// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "isLoggedIn": false,
      "setLoggedIn": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div
      className="App"
    >
      <withRouter(NavBar)
        isLoggedIn={false}
      />
      <LandingPage />
    </div>
  </Router>
</ContextProvider>
`;
